src/nvim/msgpack_rpc/channel.c

```diff
@@ -521,8 +521,19 @@ void rpc_close(Channel *channel)
   }
```

Comment:
Currently exit_event is placed on `main_loop.fast_events`. Would the problem be solved by using `main_loop.events` instead?

Reply:
Perhaps that change will solve this problem, but it will cause other problems, such as queuing an `exit_event()` while a "Press Enter or ..." prompt is displayed may result in the event not being processed and another stall occurring.